services/app-api/handlers/reports/create.ts

```diff
@@ -40,35 +42,37 @@ export const createReport = handler(async (event, _context) => {
   const requiredParams = ["reportType", "state"];
 
   // Return error if no state is passed.
-  if (!hasReportPathParams(event.pathParameters!, requiredParams)) {
+  if (
+    !event.pathParameters ||
+    !hasReportPathParams(event.pathParameters, requiredParams)
```

Comment:
I'm getting a little mixed up on this logic. So we are wanting path parameters to not exist, but also passing them into a function that checks if they do exist (to then make sure they don't)?

Reply:
The `hasReportPathParams` function checks for certain required parameters, but doesn’t affect the type inference of `event.pathParameters` . Adding an additional check for missing all path parameters entirely allows us to use fewer optionals in the handler.